Maple2.Server.DebugGame/Graphics/Resources/Texture.cs

```diff
@@ -0,0 +1,121 @@
+
+        Image<Bgra32> image = Image.Load<Bgra32>(decoderOptions, path);
+
+        Texture2DDesc textureDesc = new Texture2DDesc {
+            Width = (uint) image.Width,
+            Height = (uint) image.Height,
+            Format = Silk.NET.DXGI.Format.FormatB8G8R8A8Unorm,
+            MipLevels = 1,
+            BindFlags = (uint) BindFlag.ShaderResource,
+            Usage = Usage.Default,
+            CPUAccessFlags = 0,
+            MiscFlags = (uint) ResourceMiscFlag.None,
+            SampleDesc = new SampleDesc(1, 0),
+            ArraySize = 1
+        };
+
+        if (image.DangerousTryGetSinglePixelMemory(out var imageData)) {
+            using (var pixelData = imageData.Pin()) {
+                SubresourceData subresourceData = new SubresourceData {
+                    PSysMem = pixelData.Pointer,
+                    SysMemPitch = (uint) image.Width * sizeof(int),
+                    SysMemSlicePitch = (uint) (image.Width * sizeof(int) * image.Height)
+                };
+
+                ID3D11Texture2D* texture = default;
+                SilkMarshal.ThrowHResult(Context.DxDevice.CreateTexture2D(
+                    pDesc: in textureDesc,
+                    pInitialData: in subresourceData,
+                    ppTexture2D: ref texture));
+                Texture2D = texture;
+            }
+        } else {
+            // handle split texture memory. likely covers .dds textures
+        }
+
+        ShaderResourceViewDesc resourceViewDesc = new ShaderResourceViewDesc {
+            Format = textureDesc.Format,
+            ViewDimension = D3DSrvDimension.D3DSrvDimensionTexture2D,
+            Anonymous = new ShaderResourceViewDescUnion {
+                Texture2D = {
+                    MostDetailedMip = 0,
+                    MipLevels = 1
+                }
+            }
+        };
+
+        ID3D11ShaderResourceView* resourceView = default;
+        SilkMarshal.ThrowHResult(Context.DxDevice.CreateShaderResourceView(
+            pResource: Texture2D,
+            pDesc: ref resourceViewDesc,
+            ppSRView: ref resourceView));
+        ResourceView = resourceView;
+
+        SamplerDesc samplerDesc = new SamplerDesc {
+            Filter = Filter.MinMagMipLinear,
+            AddressU = TextureAddressMode.Clamp,
+            AddressV = TextureAddressMode.Clamp,
+            AddressW = TextureAddressMode.Clamp,
+            MipLODBias = 0,
+            MaxAnisotropy = 1,
+            MinLOD = float.MinValue,
+            MaxLOD = float.MaxValue
+        };
+
+        samplerDesc.BorderColor[0] = 0.0f;
+        samplerDesc.BorderColor[1] = 0.0f;
+        samplerDesc.BorderColor[2] = 0.0f;
+        samplerDesc.BorderColor[3] = 1.0f;
+
+        ID3D11SamplerState* samplerState = default;
+        SilkMarshal.ThrowHResult(Context.DxDevice.CreateSamplerState(
+            pSamplerDesc: in samplerDesc,
+            ppSamplerState: ref samplerState));
+        SamplerState = samplerState;
+    }
```

Comment:
Errors are already checked for with `SilkMarshal.ThrowHResult`

Reply:
If there are any failures for something like this it's better to see a loud failure to be alerted that it will be fixed, rather than sweeping it under the rug. I'll have to make more complex resource management logic later so any handling I do right now before adding that will be a moot point and get in the way of doing it properly